webapp/packages/plugin-data-viewer/src/ValuePanelPresentation/TextValue/useTextValue.ts

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from 'react';
+
+import { isResultSetContentValue } from '../../DatabaseDataModel/Actions/ResultSet/isResultSetContentValue';
+import { ResultSetEditAction } from '../../DatabaseDataModel/Actions/ResultSet/ResultSetEditAction';
+import { ResultSetFormatAction } from '../../DatabaseDataModel/Actions/ResultSet/ResultSetFormatAction';
+import { ResultSetSelectAction } from '../../DatabaseDataModel/Actions/ResultSet/ResultSetSelectAction';
+import type { IDatabaseDataModel } from '../../DatabaseDataModel/IDatabaseDataModel';
+import type { IDatabaseResultSet } from '../../DatabaseDataModel/IDatabaseResultSet';
+import { useAutoFormat } from './useAutoFormat';
+
+type Props = {
+  resultIndex: number;
+  model: IDatabaseDataModel<any, IDatabaseResultSet>;
+  currentContentType: string;
+};
+
+export function useTextValue({ model, resultIndex, currentContentType }: Props) {
+  const format = model.source.getAction(resultIndex, ResultSetFormatAction);
+  const editor = model.source.getAction(resultIndex, ResultSetEditAction);
+  const selection = model.source.getAction(resultIndex, ResultSetSelectAction);
+  const focusCell = selection.getFocusedElement();
+  const formatter = useAutoFormat();
+  const [value, setValue] = useState<string>('');
+  const firstSelectedCell = selection.elements?.[0] ?? focusCell;
+  const autoFormat = !!firstSelectedCell && !editor.isElementEdited(firstSelectedCell);
+
+  useEffect(() => {
```

Comment:
Here we probably want something like this:
```js
const cellValue = ...;
const formattedValue = useMemo(() => format(cellValue), [cellValue]);
```

It's main idea you can rewrite it. The main goal here is to sync cellValue and formattedValue and prevent cases when wrong formattedValue displayed. (For example we switched cells fast or cell value updated)

Reply:
the hook is fully synchronous right now so it has more predictable behaviour excluding fast cell switching